.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,35 @@
+name: Pull Requst
+
+on: [pull_request]
+
+jobs:
+  lint:
+    name: Lint Formatting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Lint Formatting
+        run: swift format lint --recursive --strict .
+
+  test-macos:
+    name: Test macOS using Xcode latest-stable
+    runs-on: macos-latest
```

Comment:
We won't be able to run tests on macOS for now unless we set up a way to use swift.org toolchains. This is due to that _Differentiation is not part of the Xcode default toolchains. I'd love to get this as well at some point though through something easy like swiftly working directly on macOS.

Reply:
fixed